exchanges/gateio/gateio.go

```diff
@@ -165,7 +165,7 @@ var (
 	errInvalidSettlementQuote           = errors.New("symbol quote currency does not match asset settlement currency")
 	errInvalidSettlementBase            = errors.New("symbol base currency does not match asset settlement currency")
 	errMissingAPIKey                    = errors.New("missing API key information")
-	errInvalidTextValue                 = errors.New("invalid text value, requires prefix `t-`")
+	errInvalidClientOrderIDTextPrefix   = errors.New("invalid text value, requires prefix `t-`")
```

Comment:
🐒  This feels like a hard topic, but having "OrderID" in there feels wrong, because it's not really an order ID. They say you "can set custom ID" with this field, but then the schema also contains client_order_id for transfers.
What's niggling me here is the joining of ClientOrderID and Text. I can see why you're saying that. This is super messy.
I think I'm landing on just `ClientOrderID`, and maybe changing the field we use from Text to being that too, with json being the only place we fix it up, and then comment/link to "sent as field `text` to api".
```suggestion
	errInvalidClentOrderIDPrefix   = errors.New("invalid ClientOrderID value, requires prefix `t-`")
```

Happy to have this ignored if it didn't niggle at you too.

Reply:
Originally it was mapped to `text` as per exchange API. This was a change request by @gloriousCode. He can chime in. I prefer it just being text regardless of how its being used. I think linking our unified terminology to api level fields is prone to inconsistency. We already do this "exchange" with `IBotExchange`, if I am hooking directly into API mapped functions bypassing `IBotExchange` (which I do a lot) I am diving into API docs for implementation.

Reply:
This is very involved now. I think the original var errInvalidText is fitting and I was in a mood when I all-caps the change request

Reply:
Why use a few words when the first chapter of Moby Dick would suffice?

We definitely need to talk more about this field!

Okay. Resolving.